Team6220_PowerPlay/src/main/java/org/firstinspires/ftc/team6220_PowerPlay/BaseAutonomous.java

```diff
@@ -35,92 +47,161 @@ public void driveInches(double heading, double targetDistance) {
 
             if (angleError > 180.0) {
                 angleError -= 360.0;
             } else if (angleError < -180.0) {
                 angleError += 360.0;
             }
 
-            // robot is turning counter-clockwise
-            if (angleError > 0) {
-                motorPower = Math.min(angleError / -180.0, -0.2);
-
-            // robot is turning clockwise
-            } else {
-                motorPower = Math.max(angleError / -180.0, 0.2);
-            }
+            motorPower = angleError * -Constants.TURNING_KP;
+            motorPower = Math.max(Math.min(Math.abs(motorPower), Constants.MAXIMUM_TURN_POWER_AUTONOMOUS), Constants.MINIMUM_TURN_POWER) * Math.signum(motorPower);
 
             motorFL.setPower(motorPower);
             motorFR.setPower(-motorPower);
             motorBL.setPower(motorPower);
             motorBR.setPower(-motorPower);
-
-            telemetry.addData("current", currentAngle);
-            telemetry.addData("error", angleError);
-            telemetry.addData("power", motorPower);
-            telemetry.update();
         }
 
-        motorFL.setPower(0.0);
-        motorFR.setPower(0.0);
-        motorBL.setPower(0.0);
-        motorBR.setPower(0.0);
+        stopDriveMotors();
     }
 
-    public void driveSlidesToPosition(int position) {
-        int error = position - motorLeftSlides.getCurrentPosition();
+    /**
+     * this method will allow the slides to move to a specified target position
+     * @param targetPosition target position for slides motors in ticks
+     */
+    public void driveSlidesAutonomous(int targetPosition) {
+        int error = targetPosition - motorLeftSlides.getCurrentPosition();
+        double motorPower;
+
+        // while slides aren't at target position
+        while (Math.abs(error) > Constants.ROBOT_SLIDE_TOLERANCE_TICKS && opModeIsActive()) {
+            error = targetPosition - motorLeftSlides.getCurrentPosition();
+            motorPower = error * Constants.SLIDE_MOTOR_KP;
 
-        while (Math.abs(error) > 25 && opModeIsActive()) {
+            // slides going down - full speed
             if (error < 0) {
-                if (motorLeftSlides.getCurrentPosition() > 600) {
-                    motorLeftSlides.setPower(-0.05);
-                    motorRightSlides.setPower(-0.05);
-                } else {
-                    motorLeftSlides.setPower(-1);
-                    motorRightSlides.setPower(-1);
+                motorLeftSlides.setPower(-1.0);
+                motorRightSlides.setPower(-1.0);
+                // slides going up - proportional control
+            } else {
+                motorLeftSlides.setPower(motorPower);
+                motorRightSlides.setPower(motorPower);
+            }
+        }
+
+        motorLeftSlides.setPower(Constants.SLIDE_FEEDFORWARD);
+        motorRightSlides.setPower(Constants.SLIDE_FEEDFORWARD);
+    }
+
+    // detect signal on signal sleeve
```

Comment:
This should be a javadoc comment

Reply:
Javadoc comment made, and increased comment detail